### PR TITLE
Gmailbug

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/26957

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 1]
```diff
@@ -1667,49 +1669,66 @@ def attachment_handler(message, attachments):
...
               transientFile, transientFileContent, transientFileCID, additional_headers, templateParams):
-    message = MIMEMultipart()
+    demisto.log("the body is " + str(body))
+    demisto.log("the attach_cid is " + str(attach_cid))
+    demisto.log("htmlBody is " + str(htmlBody))
+    demisto.log("entry_ids is " + str(entry_ids))
+    demisto.log("file_names is " + str(file_names))
+    if entry_ids == [] and file_names == [] and attach_cid == [] and htmlBody and body is None:
+        # if there is only htmlbody and no attachments to the mail , we would like to send it without attaching the body
+        demisto.log("entered to only html body without attachments")
```
Comment (yaakovi) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1679]: remove

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 2]
```diff
@@ -1667,49 +1669,66 @@ def attachment_handler(message, attachments):
...
+        message = MIMEText(htmlBody, 'html')
+    elif entry_ids == [] and file_names == [] and attach_cid == [] and body and htmlBody is None:
+        # if there is only body and no attachments to the mail , we would like to send it without attaching every part
+        demisto.log("entered to only body without attachments")
```
Comment (yaakovi) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1683]: remove

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 3]
```diff
@@ -1667,49 +1669,66 @@ def attachment_handler(message, attachments):
...
+                htmlBody = htmlBody.format(**templateParams)
 
-    msg = MIMEText(body, 'plain', 'utf-8')
-    message.attach(msg)
-    htmlAttachments = []  # type: list
-    inlineAttachments = []  # type: list
+        if additional_headers is not None and len(additional_headers) > 0:
+            for h in additional_headers:
+                header_name_and_value = h.split('=')
+                message[header_name_and_value[0]] = header(header_name_and_value[1])
```
Comment (yaakovi) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1708]:
```suggestion
                header_name, header_value = h.split('=')
                message[header_name] = header(header_value)
```

**Comment by orhovy** [comment]
Waiting for customer to approve this change, meanwhile opened a pr. 
All debug prints will be removed when this change will be approved by the customer

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 4]
```diff
@@ -1667,49 +1666,59 @@ def attachment_handler(message, attachments):
...
+        inlineAttachments = []  # type: list
 
-    else:
-        # if not html body, cannot attach cids in message
-        transientFileCID = None
+        if htmlBody is not None:
+            htmlBody, htmlAttachments = handle_html(htmlBody)
+            msg = MIMEText(htmlBody, 'html', 'utf-8')
+            message.attach(msg)
+            if attach_cid is not None and len(attach_cid) > 0:
```
Comment (yaakovi) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1709]:
```suggestion
            if attach_cid:
```

**Comment by orhovy** [comment]
Should I send another unified yml to the customer to check changes again?

**Pull request merged**